assets/js/blocks/cart-checkout/cart/frontend.js

```diff
@@ -2,13 +2,35 @@
  * External dependencies
  */
 import { withRestApiHydration } from '@woocommerce/block-hocs';
+import { useCollection } from '@woocommerce/base-hooks';
 
 /**
  * Internal dependencies
  */
 import FullCart from './full-cart';
+import EmptyCart from './empty-cart';
 import renderFrontend from '../../../utils/render-frontend.js';
 
+/**
+ * Wrapper component to supply API data and show empty cart view as needed.
+ */
+const CartFrontend = () => {
+	const { results: cartData, isLoading } = useCollection( {
+		namespace: '/wc/store',
+		resourceName: 'cart',
+	} );
+
+	const cartItems = isLoading ? [] : cartData.items;
+	const isCartEmpty = ! isLoading && cartItems.length <= 0;
+
+	// @todo render cart loading state e.g. animated spinner
+	return isCartEmpty ? (
+		<EmptyCart />
```

Comment:
If the cart is empty, you shouldn't need to do anything, because by default it will already render the inner blocks.

Currently with an empty cart this is what I see:

![imatge](https://user-images.githubusercontent.com/3616980/74647042-3f1e5080-517b-11ea-9d9c-9b5b6e30feaf.png)

Can you reproduce it too?

Reply:
To clarify what Albert is saying here, by default the rendered content is the inner blocks from the Empty Cart view. So there should be no need to specifically call the `EmptyCart` component.  Here's an example of how you could control what shows for the frontend rendering depending on cart contents:

```jsx
/**
 * External dependencies
 */
import { withRestApiHydration } from '@woocommerce/block-hocs';
import { useCollection } from '@woocommerce/base-hooks';
import { RawHTML } from '@wordpress/element';

/**
 * Internal dependencies
 */
import FullCart from './full-cart';
import renderFrontend from '../../../utils/render-frontend.js';

/**
 * Wrapper component to supply API data and show empty cart view as needed.
 *
 * @param EmptyCart
 */
const CartFrontend = ( { emptyCart } ) => {
	const { results: cartData, isLoading } = useCollection( {
		namespace: '/wc/store',

		resourceName: 'cart',
	} );

	const cartItems = isLoading ? [] : cartData.items;
	const isCartEmpty = ! isLoading && cartItems.length <= 0;

	// @todo render cart loading state e.g. animated spinner
	if ( isLoading ) {
		return null;
	}
	return isCartEmpty ? (
		<RawHTML>{ emptyCart }</RawHTML>
	) : (
		<FullCart cartItems={ cartItems } cartTotals={ cartData.totals } />
	);
};

const selector = '.wp-block-woocommerce-cart';

const getProps = ( el ) => {
	return {
		emptyCart: el.innerHTML,
	};
};

renderFrontend( selector, withRestApiHydration( CartFrontend ), getProps );
```

Reply:
Great – I've implemented this in 39bf43a21c2292ad38b5531bb0124fb9179ca98c, it works well :)

Note as part of this commit I removed the existing code which was removing `is-loading` class from all the `.wp-block-woocommerce-cart` elements. I'm assuming this was early "loading state" support and we can revisit this when working on #1744.

Reply:
> I removed the existing code which was removing is-loading class from all the .wp-block-woocommerce-cart elements.

Yes, that's okay and why I didn't have it in my suggested code as well. It's unneeded because `is-loading` is removed as a part of `renderFrontend`.

One thing that is missing in 39bf43a is this logic in my example:

```jsx
if ( isLoading ) {
    return null;
}
```

Without it, the full cart is shown before the empty cart component is rendered. There's no need to show anything until the cart items are resolved.  I realize you have an issue for handling loading state, but this is a three-liner that _resolves_ that _now_ and the issue can be used for just enhancing with a loading state if we think it's needed (I don't think it will be).

Reply:
Missed that! Now added here: eea85d5d6c6241842e09801f3aa535f38957bd06